File: JS/Problem 3.js
//Find the largest prime factor of a number.

function isPrime(num) {
  for (var i = 2; i < num; i++) {
    if(num % i === 0) {
      return false;
    }
  }
  return true;
}

var limit = prompt("Choose a limit");

for(i=2; i<limit; i++) {
    if(limit % i === 0 && isPrime(i)) {
        x = i;
    }
}
console.log(x);

File: Ruby/Problem 3.rb
# What is the largest prime factor of the number 600851475143 ?

def prime? num
    (2..(num-1)).each { |i| return false if num % i == 0 }
    return true
end


limit = 600851475143
(1..(limit-1)).each do |x|
    if(limit % x == 0) && (prime? x)
		puts x
	end
end